video_encoding/utils.py

```diff
@@ -0,0 +1,38 @@
+import os
+import tempfile
+
+
+def get_fieldfile_local_path(fieldfile):
+    local_temp_file = None
+
+    if hasattr(fieldfile, 'storage'):
+        storage = fieldfile.storage
+        try:
+            # Try to access with path
+            storage_local_path = storage.path(
+                fieldfile.path
+            )
+        except (NotImplementedError, AttributeError):
+            # Storage doesnt support absolute paths, download
+            # file to a temp local dir
+            storage_file = storage.open(
+                fieldfile.name, 'r'
+            )
+
+            local_temp_file = tempfile.NamedTemporaryFile(
+                delete=False
+            )
+            local_temp_file.write(storage_file.read())
+            local_temp_file.seek(0)
+
+            storage_local_path = local_temp_file.name
+    else:
+        # Its a local file with no storage abstraction
+        try:
+            path = os.path.abspath(fieldfile.path)
+        except AttributeError:
+            path = os.path.abspath(fieldfile.name)
+
+        storage_local_path = path
+
+    return storage_local_path, local_temp_file
```

Comment:
I am wondering if we could make this implementation a bit more robust by using [contextlib.contextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager). Something like

```python
@contextlib.contextmanager
def get_local_path(fieldfile: File) -> str:
    if not hasattr(fieldfile, 'storage'): 
        storage = fieldfile.storage
        try:
            # Try to access with path
            yield storage.path(fieldfile.path)
        except (NotImplementedError, AttributeError):
            # Storage doesnt support absolute paths,
            # download file to a temp local dir
            with tempfile.NamedTemporaryFile(delete=False) as temp_file:
                storage_file = storage.open(fieldfile.name, 'r')

                temp_file.write(storage_file.read())

            try:
                yield temp_file.name
            finally:
                os.unlink(temp_file.name)
    else:
        # Its a local file with no storage abstraction
        try:
            yield os.path.abspath(fieldfile.path)
        except AttributeError:
            yield os.path.abspath(fieldfile.name)
```

with this, we could simply use it as a context manager and don't need to worry about closing and unlinking the temporary file.

```python
    def _get_video_info(self):
        """
        Returns basic information about the video as dictionary.
        """
        if not hasattr(self, '_info_cache'):
            encoding_backend = get_backend()
            with get_local_path(self) as local_path:
                self._info_cache = encoding_backend.get_media_info(path)
        return self._info_cache
```

What do you think about this? :)